domain/storageprovisioning/service/service.go

```diff
@@ -181,3 +198,49 @@ func (s *Service) GetStorageResourceTagsForApplication(
 
 	return resourceTags, nil
 }
+
+// GetStorageIDsForUnit returns the storage IDs for the given unit UUID.
+//
+// The following errors may be returned:
+// - [github.com/juju/juju/core/errors.NotValid] when the provided unit UUID
+// is not valid.
+// - [github.com/juju/juju/domain/application/errors.UnitNotFound] when no
+// unit exists for the supplied unit UUID.
+// - [corestorage.InvalidStorageID] when the provided unit UUID is invalid.
+func (s *Service) GetStorageIDsForUnit(ctx context.Context, unitUUID coreunit.UUID) ([]corestorage.ID, error) {
+	if err := unitUUID.Validate(); err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	ids, err := s.st.GetStorageIDsForUnit(ctx, unitUUID.String())
```

Comment:
We have been passing the strong type to state in the rest of the domain and as recently as the last PR. What is the change in design now?

Reply:
We prefer primitives to structs in the state layer?